python-checks/src/test/resources/checks/tfFunctionDependOnOutsideVariable.py

```diff
@@ -0,0 +1,64 @@
+import tensorflow as tf
+
+foo = 3
+
+@tf.function
+def non_compliant(): # Noncompliant {{This function should not depend implicitly on a global or free variable.}}
+   #^^^^^^^^^^^^^
+    return foo + 1
+          #^^^< {{Variable used here.}}
+
+no_symbol = True
+def no_symbol(): ...
+@tf.function
+def non_compliant2(): # Noncompliant {{This function should not depend implicitly on a global or free variable.}}
+    something_else = no_symbol
+    return something_else + 1
+
+@tf.function
+def non_compliant3(): # Noncompliant
+   #^^^^^^^^^^^^^^
+    something = foo
+               #^^^< {{Variable used here.}}
+    smth = no_symbol
```

Comment:
Since `no_symbol` may be a function, do we really want it to raise an issue/be a secondary location?
I feel we shouldn't raise when we are uncertain of what the symbol represents (ideally only raising on "pure" Python variables).